docs/main/user-guides/upgrades.md

```diff
@@ -112,24 +103,53 @@ Add both of these options, if your chart has both of these enabled.
 
 2. Verify that the CRDs, Volumes, Snapshots, and StoragePools are not affected by the upgrade process.
 
+### From 4.x to 4.2
+
+This section describes the Local Storage upgrade from OpenEBS chart 4.x to OpenEBS 4.2.
+
+1. Execute the 4.2 upgrade command. 
+
+```
+helm upgrade openebs openebs/openebs -n openebs -f old-values.yaml --version 4.2
+```
+
+2. Verify that the CRDs, Volumes, Snapshots, and StoragePools are not affected by the upgrade process.
+
 ## Replicated Storage Upgrade
 
-### From 4.x to 4.1.1 or Higher
+:::note
+Downgrades are not supported.
+:::
+
+### From 3.x to 4.2
+
+This section describes the Replicated Storage upgrade from OpenEBS Umbrella chart 3.x to OpenEBS 4.2.
 
-This section describes the Replicated Storage upgrade from OpenEBS Umbrella chart 4.x to OpenEBS 4.1.1 or higher.
+1. Start the helm upgrade process with the new chart, i.e. 4.2 by using the below command:
+
+:::caution
+Upgrades from 3.x to 4.x require the option `--set mayastor.agents.core.rebuild.partial.enabled=false` in the **helm upgrade** command to ensure data consistency during the upgrade. Upgrades from 4.x onwards to newer versions do not require it.
 
-1. Start the helm upgrade process with the new chart, i.e. 4.1.1 by using the below command:
+This applies to the **kubectl mayastor upgrade** command as well, if you're using the `mayastor/mayastor` chart and not the `openebs/openebs` chart: `kubectl mayastor upgrade --set agents.core.rebuild.partial.enabled=false`
```

Comment:
We don't want to officially publish steps to support this scenario IMO.

Reply:
@avishnu Do you mean there will not be a direct upgrade path from 3.x to 4.2 going forward?

Reply:
Sorry for misunderstanding. Did you mean line 133 should be removed as umbrella chart is the only supported chart for upgrade?

Reply:
That's what I meant, yes.

Reply:
The required changes are done. :) @avishnu